src/qibolab/platforms/abstract.py

```diff
@@ -195,53 +173,125 @@ def update(self, updates: dict):
-                    if par == "drive_amplitude":
+                        self.settings["native_gates"]["single_qubit"][qubit]["MZ"]["amplitude"] = amplitude
+
+                    # rabi_amplitude / flipping
+                    if par == "drive_amplitude" or par == "amplitudes":
                         self.single_qubit_natives[qubit]["RX"]["amplitude"] = amplitude
+                        self.settings["native_gates"]["single_qubit"][qubit]["RX"]["amplitude"] = amplitude
+                        self.settings["characterization"]["single_qubit"][qubit]["pi_pulse_amplitude"] = amplitude
+
+                # rabi_duration
+                elif par == "drive_length":
+                    duration = int(value)
+                    self.single_qubit_natives[qubit]["RX"]["duration"] = duration
+                    self.settings["native_gates"]["single_qubit"][qubit]["RX"]["duration"] = duration
+
+                # ramsey
                 elif par == "t2":
-                    self.qubits[qubit].T2 = float(value)
-                elif "t1" == par:
-                    self.qubits[qubit].T1 = float(value)
-                elif "threshold" == par:
-                    self.qubits[qubit].threshold = float(value)
-                elif "iq_angle" == par:
-                    self.qubits[qubit].iq_angle = float(value)
+                    t2 = float(value)
+                    self.qubits[qubit].T2 = t2
+                    self.settings["characterization"]["single_qubit"][qubit]["T2"] = t2
+
+                # spin_echo
+                elif par == "t2_spin_echo":
+                    t2_spin_echo = float(value)
+                    self.qubits[qubit].T2_spin_echo = t2_spin_echo
+                    self.settings["characterization"]["single_qubit"][qubit]["T2_spin_echo"] = t2_spin_echo
+
+                # t1
+                elif par == "t1":
+                    t1 = float(value)
+                    self.qubits[qubit].T1 = t1
+                    self.settings["characterization"]["single_qubit"][qubit]["T1"] = t1
+
+                # classification
+                elif par == "threshold":
+                    threshold = float(value)
+                    self.qubits[qubit].thresold = threshold
+                    self.settings["characterization"]["single_qubit"][qubit]["threshold"] = threshold
+
+                # classification
+                elif par == "iq_angle":
+                    iq_angle = float(value)
+                    self.qubits[qubit].iq_angle = iq_angle
+                    self.settings["characterization"]["single_qubit"][qubit]["iq_angle"] = iq_angle
+
+                # classification
+                elif par == "mean_gnd_states":
+                    mean_gnd_states = str(value)
+                    self.qubits[qubit].iq_angle = iq_angle
+                    self.settings["characterization"]["single_qubit"][qubit]["mean_gnd_states"] = mean_gnd_states
```

Comment:
I believe this should be
```suggestion
                    self.qubits[qubit].mean_gnd_states = mean_gnd_states
                    self.settings["characterization"]["single_qubit"][qubit]["mean_gnd_states"] = mean_gnd_states
```